packages/node_modules/pouchdb-utils/src/uuid.js

```diff
@@ -39,7 +41,17 @@ var chars = (
   'abcdefghijklmnopqrstuvwxyz'
 ).split('');
 function getValue(radix) {
-  return 0 | Math.random() * radix;
+  // Using maxRandom of 255 will produce an un-even spread of random numbers.
+  // Instead, find the biggest number (within 8-bit range) that is a multiple of
+  // radix.
+  var maxRandom = 255 - (256 % radix);
+
+  var randomValue;
+  do {
+    randomValue = randomUInt8();
+  } while (randomValue > maxRandom);
+
+  return randomValue % radix;
```

Comment:
I get the nervousness! While loops based on random values are definitely cause for another look. tl;dr; though, in normal uuids it never loops, and worst case it almost never loops.

`randomUInt8` generates numbers between 0 and 255, and so the theoretical worst the threshold % chance can be is ~50% (a radix of 129). In legit use of `uuid()` with radixes under 62 the worst case (radix 52) is ~81%. In the standard UUID case where the radix is 16 it's actually *100%*: i.e. it will never loop.

The intention with this approach was to change as little as possible in regard with how you generated uuids. To match what you can do with web crypto (generate random bytes) with what you can do with Math.random (generate a float 0 < x < 1) I hid it behind generating the index into char based on the radix. The reason it can loop (on some radixes) is that if the random number is larger than the maximum multiple of the radix then we'll generate uneven random numbers (because the entire collection of possibilities now don't map evenly against every radix possibility).

A faster approach (in terms of UUID generation)--or at least one that doesn't have a while loop in it-- would be to just use the `node-uuid` library. I'm not sure how it's MIT licence gels with PouchDB's Apache one, but if it's acceptable we could pretty much just copy https://github.com/kelektiv/node-uuid/blob/master/v4.js whole cloth.